src/KamadaKawai.jl

```diff
@@ -0,0 +1,87 @@
+import Optim
+using Distances
+function kamada_kawai_layout(G, X=nothing; C= 1.0, MAXITER=100 )
+    Offset = 0.0
+    if X===nothing
+        locs_x = zeros(nv(G))
+        locs_y = zeros(nv(G))
+    else
+        locs_x = X[1]
+        locs_y = X[2]
+    end
+
+    function Objective(M,K)
+        D = pairwise(Euclidean(),M, dims=2)
+        D-= K
+        R = sum(D.^2)/2
+        return R
+    end
+
+    function dObjective!(dR,M,K)
+        dR .= zeros(size(M))
+        Vs = size(M,2)
+        D = pairwise(Euclidean(),M, dims=2)
+        D += I # Prevent division by zero
+        D .= K./D # Use negative for simplicity, since diag K = 0 everything is fine.
+        D .-= 1.0 # (K-(D+I))./(D+I) = K./(D+I) .- 1.0
+        D += I # Remove the false diagonal
+        for v1 in 1:Vs
+            dR[:,v1] .= -M[:,v1]*sum(D[:,v1])
+        end
+        dR .+= M*D
+        dR .*=2
+        return dR
+    end
+
+    function scaler(z, a, b)
+        2.0*((z - a)/(b - a)) - 1.0
+    end
+    # Stack individual graphs next to each other
+    for SubGraphVertices in connected_components(G)
+        SubGraph = induced_subgraph(G,SubGraphVertices)[1]
+        N = nv(SubGraph)
+        if X !== nothing
+            _locs_x = locs_x[SubGraphVertices]
+            _locs_y = locs_y[SubGraphVertices]
+        else
+            Vertices=collect(vertices(SubGraph))
+            Vmax=findmax([degree(SubGraph,x) for x in vertices(SubGraph)])[2]
+            filter!(x->x!=Vmax, Vertices)
```

Comment:
Maybe you could use `filter!( !isequal(Vmax), Vertices)` here or ``filter!( !(==(Vmax)), Vertices)`

Reply:
No problem.